app/forms/mappings/Formatters.scala

```diff
@@ -222,4 +222,35 @@ trait Formatters extends Transforms {
 
     }
 
+  private[mappings] def mandatoryGIINFormatter(requiredKey: String,
+                                               lengthKey: String,
+                                               invalidKey: String,
+                                               formatKey: String,
+                                               regex: String,
```

Comment:
I'm not convinced the regexs should be parameters - do we expect to use this function more than once and using different regexs?

To my mind the validation of a GIIN is fixed and we should inline the regexs into this function.

Reply:
I'd say it keeps it consistent having all regex formulas we use sat in the same place but if thats not best practice then this can be changed? I don't really see the difference.

Reply:
I'm not a fan of having all the regex in one file just for the sake of it, but at least it's consistent so I'm kinda ok with that.

However, I do think this function should only take the error keys as parameters. I don't see why we would want to pass the regex as a parameter? As I say:

> do we expect to use this function more than once and using different regexs?

I would be inclined to import `import utils.RegexConstants` in this file and reference the regexs directly, rather than having them as a parameter to the function?

Reply:
Happy with that approach i've just amended and implemented this^